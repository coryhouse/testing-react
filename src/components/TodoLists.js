import React from 'react';
import TodoItem from './TodoItem';

const TodoLists = props => {
  return (
    <div className="todo-list">
      {props.todos.length === 0 && <p>No todos found. Please add some.</p>}
      {props.todos.map((todo, index) => (
        <TodoItem key={index} todo={todo}></TodoItem>
      ))}
    </div>
  );
};

export default TodoLists;
