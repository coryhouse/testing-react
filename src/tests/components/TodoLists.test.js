import React from 'react';
import { shallow } from 'enzyme';
import TodoLists from '../../components/TodoLists';

test('should test TodoLists component with default state of empty array', () => {
  const wrapper = shallow(<TodoLists todos={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should test TodoLists component with list of todos', () => {
  const wrapper = shallow(
    <TodoLists todos={['Read Newspaper', 'Publish Article']} />
  );
  expect(wrapper).toMatchSnapshot();
});
