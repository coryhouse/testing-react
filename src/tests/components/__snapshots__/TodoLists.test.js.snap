// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test TodoLists component with default state of empty array 1`] = `
<div
  className="todo-list"
>
  <p>
    No todos found. Please add some.
  </p>
</div>
`;

exports[`should test TodoLists component with list of todos 1`] = `
<div
  className="todo-list"
>
  <TodoItem
    key="0"
    todo="Read Newspaper"
  />
  <TodoItem
    key="1"
    todo="Publish Article"
  />
</div>
`;
